Celeste.Mod.mm/Patches/Monocle/Tracker.cs

```diff
@@ -123,5 +123,49 @@ private static List<Type> GetSubclasses(Type type) {
             // don't hold references to all the types anymore
             _temporaryAllTypes = null;
         }
+
+        public static void AddEntityToTracker(Type type, params Type[] subTypes) {
+            StoredEntityTypes.Add(type);
+            if (!TrackedEntityTypes.TryGetValue(type, out List<Type> value)) {
+                TrackedEntityTypes[type] = new List<Type> { type };
+            } else if (!value.Contains(type)) {
+                value.Add(type);
+            }
+            foreach (Type subType in subTypes) {
+                if (!TrackedEntityTypes.TryGetValue(subType, out List<Type> subvalue)) {
+                    TrackedEntityTypes[subType] = new List<Type> { type };
+                } else if (!subvalue.Contains(type)) {
+                    subvalue.Add(type);
+                }
+            }
+            Dictionary<Type, List<Entity>> entities = Engine.Scene?.Tracker.Entities;
+            if (entities != null && !entities.ContainsKey(type)) {
+                entities[type] = Engine.Scene.Entities.Where((e) => e.GetType() == type).ToList();
```

Comment:
This doesn't handle inheritance, which could lead to weird edge cases between first room loads and subsequent ones